### ISSUE TITLE
Installation fails if drive have already a Kairos install

### ISSUE THREAD

**Issue by alexander-bauer** [kairos-io/kairos#499]
**Kairos version:**

```
NAME="kairos-core-opensuse"
VERSION="v0.0.0-50d9143"
ID="kairos"
ID_LIKE="kairos-core-opensuse"
VERSION_ID="v0.0.0-50d9143"
PRETTY_NAME="kairos-core-opensuse v0.0.0-50d9143"
ANSI_COLOR="0;32"
BUG_REPORT_URL="https://github.com/kairos-io/kairos/issues"
HOME_URL="https://github.com/kairos-io/kairos"
IMAGE_REPO="quay.io/kairos/core-opensuse"
IMAGE_LABEL="latest"
GITHUB_REPO="kairos-io/kairos"
VARIANT="core"
FLAVOR="opensuse"
```

**CPU architecture, OS, and Version:**

```
Linux localhost 5.14.21-150400.24.33-default #1 SMP PREEMPT_DYNAMIC Fri Nov 4 13:55:06 UTC 2022 (76cfe60) x86_64 x86_64 x86_64 GNU/Linux
```

**Describe the bug**

I discovered this attempting to install the OpenSUSE variant using an ISO installer on an Atomic Pi, which is an amd64-based single board computer. There are three storage devices in play: an SD card containing the ISO (`/dev/sdb`), built-in flash memory, which should not be used (`/dev/mmcblk0*`), and the target install device, a 1TB HDD (`/dev/sda`).

**Note**: This hard drive has been partitioned before. I think this is a possible cause of the issue.

During install, the console displays:
```
INFO[<datetime>] Starting elemental version v0.0.1
INFO[<datetime>] Install called
INFO[<datetime>] Partitioning device...
ERRO[<datetime>] Failed creating new partition table: Error: Partion(s) 1, 2, 3, 4, 5 on /dev/sda have been written, but we have been unable to informt he kernel of the change, probably because it/they are in use. As a result, the old partition(s) will remain in use. You should reboot now before making further changes.
Error: 1 error occurred:
         * exit status 1

exit status 1
```

**To Reproduce**

Using a device with an attached already-partitioned hard drive, attempt to install using an OpenSUSE base with an ISO.

**Expected behavior**

With `install.device` set explicitly and `install.auto = true` in the `cloudConfig`, I expect the target hard drive partitions to be wiped, and the install to happen with no further interaction.

**Logs**

Attached is the entire output from `sudo journalctl -eb`.
[journal.txt](https://github.com/kairos-io/kairos/files/10094879/journal.txt)

**Additional context**

This is the output from `lsblk`.

```
NAME         MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
loop0          7:0    0 830.6M  1 loop /run/rootfsbase
sda            8:0    0 931.5G  0 disk 
|-sda1         8:1    0    64M  0 part 
|-sda2         8:2    0    64M  0 part 
|-sda3         8:3    0     8G  0 part 
|-sda4         8:4    0    15G  0 part 
`-sda5         8:5    0 908.4G  0 part 
sdb            8:16   1  14.9G  0 disk 
|-sdb1         8:17   1 933.2M  0 part /run/initramfs/live
`-sdb2         8:18   1     4M  0 part 
mmcblk0      179:0    0  14.6G  0 disk 
mmcblk0boot0 179:8    0     4M  1 disk 
mmcblk0boot1 179:16   0     4M  1 disk 
```

My customized ISO was built using OSBuilder with this OSArtifact definition.

```
resource "kubectl_manifest" "kairos_opensuse_api_iso" {
  yaml_body = yamlencode({
    apiVersion = "build.kairos.io/v1alpha1"
    kind       = "OSArtifact"
    metadata = {
      name      = "opensuse-api-iso"
      namespace = one(kubernetes_namespace.kairos.metadata).name
    }
    spec = {
      imageName = "quay.io/kairos/core-opensuse:latest"
      bundles = [
        "quay.io/kairos/packages:k3s-bundle-1.24.2"
      ]
      iso         = true
      cloudConfig = <<-EOF
        #cloud-config
        hostname: "api-{{ trunc 4 .MachineID }}"
        users:
          - name: "kairos"
            passwd: "kairos"
            ssh_authorized_keys:
              - "${file("${path.module}/id_rsa.pub")}"
        install:
          device: "/dev/sda"
          reboot: true
          poweroff: false
          auto: true # Required, for automated installations
          bundles:
            - quay.io/kairos/packages:k3s-bundle-1.24.2
        k3s-agent:
          enabled: true
          env:
            K3S_URL: "<redacted>
            K3S_TOKEN: "<redacted>"
      EOF
    }
  })
  # OSBuilder does not detect changes to OSArtifact manifests, so we delete and recreate.
  force_new = true
}
```

**Comment by alexander-bauer** [comment]
I've confirmed that this is not present with my hardware when I wipe the partition table ahead of time, so that the disk has no partitions when it is initially booted. I would guess that you could put this in the `initramfs` stage, and it would work. I'll try to confirm that later on: `dd if=/dev/zero of=/dev/sda bs=512 count=100` (where `/dev/sda` is the install device).

**Comment by mudler** [comment]
Good catch, that seems indeed a bug and needs investigation - I remember that was kinda of tricky as probably some mount might be happening in the boot that makes the installer fail. The installer used to call `wipefs` on the disk before starting installation.

**Comment by Itxaka** [comment]
was able to reproduce.

 - installed kairos-opensuse-70903fd-k3s.iso (fresh from master)
 - reboot
 - installed again (interactive install)

**Comment by Itxaka** [comment]
something strange that I found is that before the partitioning appears, on failed runs, there is a mount triggered. This looks similar to what elemental does internally (uses the transient mount lib from kubernetes project) but as far as I remember there should be nothing being mounted at that point, and its only afterwards when the mount stuff happens, once the partition table is created, then we create partitions and mount them.

```
Jan 19 10:53:49 localhost systemd[1]: Started Kubernetes systemd probe.
Jan 19 10:53:49 localhost systemd[1]: run-rf1ac2d07f9a6486baa1353422b78b43a.scope: Deactivated successfully.
Jan 19 10:54:00 localhost systemd[1]: kairos-interactive.service: Main process exited, code=exited, status=1/FAILURE
Jan 19 10:54:00 localhost systemd[1]: kairos-interactive.service: Failed with result 'exit-code'.
Jan 19 10:54:00 localhost systemd[1]: Failed to start kairos interactive-installer.
```

Otherwise Im not seeing anything that could be using the partition. 

There is also another thing, in which I can seem to trigger it a 100% of the time. With interactive install and doing it as fast as I can, it triggers.

If I wait a bit, it doesnt trigger. I can see in the services that the network adn boot stages havent been triggered yet when it fails, due to the metadata probing  taking a lot of time. so maybe its related to that?

**Comment by Itxaka** [comment]
There was even an attempt to force close all files but didnt work IIRC: https://github.com/davidcassany/linuxkit/pull/5

**Comment by Itxaka** [comment]
Yep, again retested like a hundred times. removing the cdrom provider makes it work 100% of the time. Enabling it back again makes it fail from time to time, so its not closing the opened devices properly somehow. what a pita.

**Comment by Itxaka** [comment]
weirdly enough, this may be because the metadata is running at the same time that elemental-cli is doing the disk management, because if you wait until the metadata plugin has run, then you cannot trigger it....weird

**Comment by santhoshdaivajna** [comment]
I hit a similar issue while trying to upgrade from kairos base image 
quay.io/kairos/core-alpine:v1.1.6  to quay.io/kairos/core-alpine-opensuse-leap:v1.4.0
Let me know, if you want me create a separate issue.


<img width="1474" alt="Screenshot 2023-01-20 at 2 54 47 PM" src="https://user-images.githubusercontent.com/504862/213660831-7ad0af32-5301-46ee-b04b-d6005807667f.png">

Later when elemental upgrade was called
<img width="1516" alt="Screenshot 2023-01-20 at 2 56 29 PM" src="https://user-images.githubusercontent.com/504862/213661166-76bd10f8-9d18-4925-ad18-1bc622e52f2e.png">

**Comment by Itxaka** [comment]
systemd log of a failure. You can see that the installer is running while the metadata plugin is still ongoing.


```
Jan 20 09:45:20 localhost elemental[1618]: INFO[2023-01-20T09:45:20Z] Running stage: network
Jan 20 09:45:20 localhost elemental[1618]: INFO[2023-01-20T09:45:20Z] Executing /system/oem/00_datasource.yaml
Jan 20 09:45:20 localhost elemental[1618]: INFO[2023-01-20T09:45:20Z] Applying 'Datasource handling' for stage 'network'. Total stages: 3
Jan 20 09:45:20 localhost elemental[1618]: INFO[2023-01-20T09:45:20Z] Processing stage step 'Pull data from provider'. ( commands: 0, files: 0, ... )
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="cdrom devices to be checked: [/dev/sr0 /dev/sr1]"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="block devices found: [/sys/class/block/loop0 /sys/class/block/loop1 /sys/class/block/loop2 /sys/class/block/loop3 /sys/class/block/loop4 /sys/class/block/loop5 /sys/class/block/loop6 /sys/class/block/loop7 /sys/class/block/sr0 /sys/class/block/sr1 /sys/class/block/vda /sys/class/block/vda1 /sys/class/block/vda2 /sys/class/block/vda3 /sys/class/block/vda4 /sys/class/block/vda5]"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="ignoring loop or ram device: loop0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="ignoring loop or ram device: loop1"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="ignoring loop or ram device: loop2"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="ignoring loop or ram device: loop3"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="ignoring loop or ram device: loop4"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="ignoring loop or ram device: loop5"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="ignoring loop or ram device: loop6"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="ignoring loop or ram device: loop7"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="checking device: /dev/sr0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): start"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): block device"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): logical block size 2048, physical block size 2048"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying fat32"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="fat32 failed: blocksize for FAT32 must be either 512 bytes or 0, not 2048"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying iso9660 with physical block size 0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="found trimmed filesystem label for device: /dev/sr0: 'COS_LIVE'"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="checking device: /dev/sr1"
Jan 20 09:45:20 localhost systemd[1]: Started Open-iSCSI.
░░ Subject: A start job for unit iscsid.service has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit iscsid.service has finished successfully.
░░ 
░░ The job identifier is 265.
Jan 20 09:45:20 localhost systemd[1]: Reached target Preparation for Remote File Systems.
░░ Subject: A start job for unit remote-fs-pre.target has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit remote-fs-pre.target has finished successfully.
░░ 
░░ The job identifier is 266.
Jan 20 09:45:20 localhost systemd[1]: Starting Login and scanning of iSCSI devices...
░░ Subject: A start job for unit iscsi.service has begun execution
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit iscsi.service has begun execution.
░░ 
░░ The job identifier is 249.
Jan 20 09:45:20 localhost iscsiadm[1630]: iscsiadm: No records found
Jan 20 09:45:20 localhost iscsiadm[1631]: iscsiadm: No records found
Jan 20 09:45:20 localhost iscsiadm[1632]: iscsiadm: Could not get list of targets from firmware. (err 21)
Jan 20 09:45:20 localhost systemd[1]: Finished Login and scanning of iSCSI devices.
░░ Subject: A start job for unit iscsi.service has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit iscsi.service has finished successfully.
░░ 
░░ The job identifier is 249.
Jan 20 09:45:20 localhost systemd[1]: Reached target Remote File Systems.
░░ Subject: A start job for unit remote-fs.target has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit remote-fs.target has finished successfully.
░░ 
░░ The job identifier is 248.
Jan 20 09:45:20 localhost systemd[1]: Starting Permit User Sessions...
░░ Subject: A start job for unit systemd-user-sessions.service has begun execution
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit systemd-user-sessions.service has begun execution.
░░ 
░░ The job identifier is 241.
Jan 20 09:45:20 localhost systemd[1]: Finished Permit User Sessions.
░░ Subject: A start job for unit systemd-user-sessions.service has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit systemd-user-sessions.service has finished successfully.
░░ 
░░ The job identifier is 241.
Jan 20 09:45:20 localhost systemd[1]: Started Serial Getty on ttyS0.
░░ Subject: A start job for unit serial-getty@ttyS0.service has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit serial-getty@ttyS0.service has finished successfully.
░░ 
░░ The job identifier is 231.
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="failed to open device read-only: /dev/sr1: Could not open device /dev/sr1 exclusively for writing"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="checking device: /dev/vda"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): start"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): block device"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): logical block size 512, physical block size 512"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying fat32"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="fat32 failed: Error reading MS-DOS Boot Sector: Could not read FAT32 BIOS Parameter Block from boot sector: Could not read embedded DOS 3.31 BPB: Error reading embedded DOS 2.0 BPB: Invalid sector size 0 provided in DOS 2.0 BPB. Must be 512"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying iso9660 with physical block size 0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="iso9660 failed: Error reading Volume Descriptor: Mismatched ISO identifier in Volume Descriptor. Found 0 expected 4344303031"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="failed to get filesystem on partition 0 for device: /dev/vda: Unknown filesystem on partition 0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="checking device: /dev/vda1"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): start"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): block device"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): logical block size 512, physical block size 512"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying fat32"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="fat32 failed: Error reading MS-DOS Boot Sector: Could not read FAT32 BIOS Parameter Block from boot sector: Invalid FAT32 version found: 17168"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying iso9660 with physical block size 0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="iso9660 failed: Error reading Volume Descriptor: Mismatched ISO identifier in Volume Descriptor. Found 0 expected 4344303031"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="failed to get filesystem on partition 0 for device: /dev/vda1: Unknown filesystem on partition 0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="checking device: /dev/vda2"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): start"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): block device"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): logical block size 512, physical block size 512"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying fat32"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="fat32 failed: Error reading MS-DOS Boot Sector: Could not read FAT32 BIOS Parameter Block from boot sector: Could not read embedded DOS 3.31 BPB: Error reading embedded DOS 2.0 BPB: Invalid sector size 0 provided in DOS 2.0 BPB. Must be 512"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying iso9660 with physical block size 0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="iso9660 failed: Error reading Volume Descriptor: Mismatched ISO identifier in Volume Descriptor. Found 2000002060 expected 4344303031"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="failed to get filesystem on partition 0 for device: /dev/vda2: Unknown filesystem on partition 0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="checking device: /dev/vda3"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): start"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): block device"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): logical block size 512, physical block size 512"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying fat32"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="fat32 failed: Error reading MS-DOS Boot Sector: Could not read FAT32 BIOS Parameter Block from boot sector: Could not read embedded DOS 3.31 BPB: Error reading embedded DOS 2.0 BPB: Invalid sector size 0 provided in DOS 2.0 BPB. Must be 512"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying iso9660 with physical block size 0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="iso9660 failed: Error reading Volume Descriptor: Mismatched ISO identifier in Volume Descriptor. Found 8000000880 expected 4344303031"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="failed to get filesystem on partition 0 for device: /dev/vda3: Unknown filesystem on partition 0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="checking device: /dev/vda4"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): start"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): block device"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): logical block size 512, physical block size 512"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying fat32"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="fat32 failed: Error reading MS-DOS Boot Sector: Could not read FAT32 BIOS Parameter Block from boot sector: Could not read embedded DOS 3.31 BPB: Error reading embedded DOS 2.0 BPB: Invalid sector size 0 provided in DOS 2.0 BPB. Must be 512"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying iso9660 with physical block size 0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="iso9660 failed: Error reading Volume Descriptor: Mismatched ISO identifier in Volume Descriptor. Found 8000000880 expected 4344303031"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="failed to get filesystem on partition 0 for device: /dev/vda4: Unknown filesystem on partition 0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="checking device: /dev/vda5"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): start"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): block device"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="initDisk(): logical block size 512, physical block size 512"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying fat32"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="fat32 failed: Error reading MS-DOS Boot Sector: Could not read FAT32 BIOS Parameter Block from boot sector: Could not read embedded DOS 3.31 BPB: Error reading embedded DOS 2.0 BPB: Invalid sector size 0 provided in DOS 2.0 BPB. Must be 512"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="trying iso9660 with physical block size 0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="iso9660 failed: Error reading Volume Descriptor: Mismatched ISO identifier in Volume Descriptor. Found 8000000880 expected 4344303031"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="failed to get filesystem on partition 0 for device: /dev/vda5: Unknown filesystem on partition 0"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="CIDATA devices to be checked: []"
Jan 20 09:45:20 localhost elemental[1618]: time="2023-01-20T09:45:20Z" level=debug msg="unique devices to be checked: [/dev/sr0 /dev/sr1]"
Jan 20 09:45:22 localhost systemd[1]: Started Kubernetes systemd probe.
░░ Subject: A start job for unit run-rc7d94064ee184f11944e7f3d3d709fdd.scope has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit run-rc7d94064ee184f11944e7f3d3d709fdd.scope has finished successfully.
░░ 
░░ The job identifier is 624.
Jan 20 09:45:22 localhost systemd[1]: run-rc7d94064ee184f11944e7f3d3d709fdd.scope: Deactivated successfully.
░░ Subject: Unit succeeded
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ The unit run-rc7d94064ee184f11944e7f3d3d709fdd.scope has successfully entered the 'dead' state.
Jan 20 09:45:25 localhost systemd[1]: Created slice User Slice of UID 0.
░░ Subject: A start job for unit user-0.slice has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit user-0.slice has finished successfully.
░░ 
░░ The job identifier is 696.
Jan 20 09:45:25 localhost systemd[1]: Starting User Runtime Directory /run/user/0...
░░ Subject: A start job for unit user-runtime-dir@0.service has begun execution
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit user-runtime-dir@0.service has begun execution.
░░ 
░░ The job identifier is 695.
Jan 20 09:45:25 localhost systemd-logind[1428]: New session 1 of user root.
░░ Subject: A new session 1 has been created for user root
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ Documentation: sd-login(3)
░░ 
░░ A new session with the ID 1 has been created for the user root.
░░ 
░░ The leading process of the session is 1634.
Jan 20 09:45:25 localhost systemd[1]: Finished User Runtime Directory /run/user/0.
░░ Subject: A start job for unit user-runtime-dir@0.service has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit user-runtime-dir@0.service has finished successfully.
░░ 
░░ The job identifier is 695.
Jan 20 09:45:25 localhost systemd[1]: Starting User Manager for UID 0...
░░ Subject: A start job for unit user@0.service has begun execution
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit user@0.service has begun execution.
░░ 
░░ The job identifier is 628.
Jan 20 09:45:25 localhost systemd[1666]: pam_unix(systemd-user:session): session opened for user root by (uid=0)
Jan 20 09:45:26 localhost systemd[1666]: Queued start job for default target Main User Target.
Jan 20 09:45:26 localhost systemd[1666]: Created slice User Application Slice.
░░ Subject: A start job for unit UNIT has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit UNIT has finished successfully.
░░ 
░░ The job identifier is 6.
Jan 20 09:45:26 localhost systemd[1666]: Started Daily Cleanup of User's Temporary Directories.
░░ Subject: A start job for unit UNIT has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit UNIT has finished successfully.
░░ 
░░ The job identifier is 10.
Jan 20 09:45:26 localhost systemd[1666]: Reached target Paths.
░░ Subject: A start job for unit UNIT has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit UNIT has finished successfully.
░░ 
░░ The job identifier is 13.
Jan 20 09:45:26 localhost systemd[1666]: Reached target Timers.
░░ Subject: A start job for unit UNIT has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit UNIT has finished successfully.
░░ 
░░ The job identifier is 9.
Jan 20 09:45:26 localhost systemd[1666]: Starting D-Bus User Message Bus Socket...
░░ Subject: A start job for unit UNIT has begun execution
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit UNIT has begun execution.
░░ 
░░ The job identifier is 12.
Jan 20 09:45:26 localhost systemd[1666]: Starting Create User's Volatile Files and Directories...
░░ Subject: A start job for unit UNIT has begun execution
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit UNIT has begun execution.
░░ 
░░ The job identifier is 5.
Jan 20 09:45:26 localhost systemd[1666]: Finished Create User's Volatile Files and Directories.
░░ Subject: A start job for unit UNIT has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit UNIT has finished successfully.
░░ 
░░ The job identifier is 5.
Jan 20 09:45:26 localhost systemd[1666]: Listening on D-Bus User Message Bus Socket.
░░ Subject: A start job for unit UNIT has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit UNIT has finished successfully.
░░ 
░░ The job identifier is 12.
Jan 20 09:45:26 localhost systemd[1666]: Reached target Sockets.
░░ Subject: A start job for unit UNIT has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit UNIT has finished successfully.
░░ 
░░ The job identifier is 11.
Jan 20 09:45:26 localhost systemd[1666]: Reached target Basic System.
░░ Subject: A start job for unit UNIT has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit UNIT has finished successfully.
░░ 
░░ The job identifier is 4.
Jan 20 09:45:26 localhost systemd[1666]: Reached target Main User Target.
░░ Subject: A start job for unit UNIT has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit UNIT has finished successfully.
░░ 
░░ The job identifier is 3.
Jan 20 09:45:26 localhost systemd[1666]: Startup finished in 182ms.
░░ Subject: User manager start-up is now complete
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ The user manager instance for user 0 has been started. All services queued
░░ for starting have been started. Note that other services might still be starting
░░ up or be started at any later time.
░░ 
░░ Startup of the manager took 182519 microseconds.
Jan 20 09:45:26 localhost systemd[1]: Started User Manager for UID 0.
░░ Subject: A start job for unit user@0.service has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit user@0.service has finished successfully.
░░ 
░░ The job identifier is 628.
Jan 20 09:45:26 localhost systemd[1]: Started Session 1 of User root.
░░ Subject: A start job for unit session-1.scope has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit session-1.scope has finished successfully.
░░ 
░░ The job identifier is 698.
Jan 20 09:45:26 localhost login[1634]: pam_unix(login:session): session opened for user root by LOGIN(uid=0)
Jan 20 09:45:26 localhost login[1634]: DIALUP AT ttyS0 BY root
Jan 20 09:45:26 localhost login[1634]: ROOT LOGIN ON ttyS0
Jan 20 09:45:28 localhost systemd[1]: Created slice Slice /system/getty.
░░ Subject: A start job for unit system-getty.slice has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit system-getty.slice has finished successfully.
░░ 
░░ The job identifier is 770.
Jan 20 09:45:28 localhost systemd[1]: Started Getty on tty2.
░░ Subject: A start job for unit getty@tty2.service has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit getty@tty2.service has finished successfully.
░░ 
░░ The job identifier is 769.
Jan 20 09:45:33 localhost systemd[1]: kairos-interactive.service: Main process exited, code=exited, status=1/FAILURE
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ An ExecStart= process belonging to unit kairos-interactive.service has exited.
░░ 
░░ The process' exit code is 'exited' and its exit status is 1.
Jan 20 09:45:33 localhost systemd[1]: kairos-interactive.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ The unit kairos-interactive.service has entered the 'failed' state with result 'exit-code'.
Jan 20 09:45:33 localhost systemd[1]: Failed to start kairos interactive-installer.
░░ Subject: A start job for unit kairos-interactive.service has failed
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit kairos-interactive.service has finished with a failure.
░░ 
░░ The job identifier is 236 and the job result is failed.
Jan 20 09:45:36 localhost login[1711]: pam_systemd(login:session): Acquired user record does not have a UID.
Jan 20 09:45:36 localhost login[1711]: pam_unix(login:session): session opened for user kairos by LOGIN(uid=0)
Jan 20 09:45:36 localhost login[1711]: LOGIN ON tty2 BY kairos
Jan 20 09:45:50 localhost systemd-timesyncd[1373]: Initial synchronization to time server 51.195.26.117:123 (2.suse.pool.ntp.org).
Jan 20 09:45:54 localhost sudo[1760]:   kairos : TTY=tty2 ; PWD=/home/kairos ; USER=root ; COMMAND=/usr/bin/journalctl
Jan 20 09:45:54 localhost systemd-logind[1428]: New session 3 of user root.
░░ Subject: A new session 3 has been created for user root
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ Documentation: sd-login(3)
░░ 
░░ A new session with the ID 3 has been created for the user root.
░░ 
░░ The leading process of the session is 1760.
Jan 20 09:45:54 localhost systemd[1]: Started Session 3 of User root.
░░ Subject: A start job for unit session-3.scope has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit session-3.scope has finished successfully.
░░ 
░░ The job identifier is 883.
Jan 20 09:45:54 localhost sudo[1760]: pam_unix(sudo:session): session opened for user root by kairos(uid=65535)
Jan 20 09:46:00 localhost elemental[1618]: 2023/01/20 09:46:00 Scaleway: Could not contact metadata service: Get "http://169.254.42.42/conf": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
Jan 20 09:46:04 localhost elemental[1618]: 2023/01/20 09:46:04 Metaldata: Probing...
Jan 20 09:46:06 localhost elemental[1618]: ERRO[2023-01-20T09:46:06Z] No metadata/userdata found. Bye
Jan 20 09:46:06 localhost elemental[1618]: WARN[2023-01-20T09:46:06Z] Error 'Skipping stage (if statement error: failed to run [ -f /oem/userdata ] && [ -f /oem/.userdata_load ]: exit status 1)' in stage name: Datasource handling stage: network
```

**Comment by Itxaka** [comment]
Patches have been merged all along the line up to our elemental-cli but until we push a new version and use that on kairos we wont avoid this.

**Comment by mudler** [comment]
@Itxaka can we close it then for now? fix should be on its way in the new releases, correct?

**Comment by Itxaka** [comment]
sure, but we are still nowhere close to release a newer elemental-cli that gets used by this so not sure if we should close it as it may happen again to someone and try to report it again?

**Comment by Itxaka** [comment]
elemental-cli has been bumped on master, this should not happen once 1.6 is released

**Comment by mudler** [comment]
seems this is now hitting v2.4.3-rc3

![image (9)](https://github.com/kairos-io/kairos/assets/2420543/372c0c47-37a6-42ad-a2d8-0fbacdb27591)

**Comment by mauromorales** [comment]
I see we brought back the elemental-cli package https://github.com/kairos-io/packages/pull/449, but the repo is archived so it's not clear where we stand here

**Comment by santhoshdaivajna** [comment]
![image (11)](https://github.com/kairos-io/kairos/assets/504862/fdd04314-1fe7-45fe-b158-31aa12ec0f43)
![image (12)](https://github.com/kairos-io/kairos/assets/504862/c9111b7e-ee34-4e44-932a-6fccb707f12d)
![image (13)](https://github.com/kairos-io/kairos/assets/504862/434ebb30-6652-4580-a8e0-0d98da832100)
blkid, lsblk, fdisk

**Comment by Itxaka** [comment]
we should probably check in the agent if the target device is mounted somewhere and fail with a new error saying that the device is in use.
